[CI] MaxDocsLimitIT.testMaxDocsLimit

### CI Link

https://gradle-enterprise.elastic.co/s/ttoy2fz7vpo3a/tests/:server:internalClusterTest/org.elasticsearch.index.engine.MaxDocsLimitIT/testMaxDocsLimit

### Repro line

`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.engine.MaxDocsLimitIT.testMaxDocsLimit" -Dtests.seed=87A99F29C5284169 -Dtests.locale=es-PR -Dtests.timezone=America/Mazatlan -Druntime.java=17`

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe/Berlin&tests.container=org.elasticsearch.index.engine.MaxDocsLimitIT&tests.test=testMaxDocsLimit#

### Failure excerpt

```
org.elasticsearch.ResourceAlreadyExistsException: index [test/JWxXBgp8T0CNYCD7JwrRlw] already exists	
	at __randomizedtesting.SeedInfo.seed([87A99F29C5284169:39D24887ACAF7B57]:0)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.validateIndexName(MetadataCreateIndexService.java:175)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.validate(MetadataCreateIndexService.java:1360)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:343)	
	at org.elasticsearch.cluster.metadata.MetadataCreateIndexService$1.execute(MetadataCreateIndexService.java:298)	
	at org.elasticsearch.cluster.service.MasterService$UnbatchedExecutor.execute(MasterService.java:551)	
	at org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1052)	
	at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1017)	
	at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:278)	
	at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:170)	
	at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:110)	
	at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:148)	
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:850)	
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:257)	
	at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:223)	
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)	
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)	
	at java.lang.Thread.run(Thread.java:833)	
```

```
  1> org.elasticsearch.cluster.coordination.FailedToCommitClusterStateException: publication failed
  1>    at org.elasticsearch.cluster.coordination.Coordinator$CoordinatorPublication$4.onFailure(Coordinator.java:1877) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.ListenableFuture.notifyListenerDirectly(ListenableFuture.java:115) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.ListenableFuture.addListener(ListenableFuture.java:55) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Coordinator$CoordinatorPublication.onCompletion(Coordinator.java:1784) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Publication.onPossibleCompletion(Publication.java:113) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Publication.onPossibleCommitFailure(Publication.java:164) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Publication$PublicationTarget$PublishResponseHandler.onFailure(Publication.java:372) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Coordinator$6.onFailure(Coordinator.java:1519) ~[main/:?]
  1>    at org.elasticsearch.action.ActionRunnable.onFailure(ActionRunnable.java:92) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.onFailure(ThreadContext.java:900) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:28) ~[main/:?]
  1>    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
  1>    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
  1>    at java.lang.Thread.run(Thread.java:1589) ~[?:?]
  1> Caused by: java.lang.IllegalArgumentException: number of documents in the index cannot exceed 12
  1>    at org.apache.lucene.index.DocumentsWriterPerThread.reserveOneDoc(DocumentsWriterPerThread.java:205) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.apache.lucene.index.DocumentsWriterPerThread.updateDocuments(DocumentsWriterPerThread.java:239) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.apache.lucene.index.DocumentsWriter.updateDocuments(DocumentsWriter.java:432) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1533) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.apache.lucene.index.IndexWriter.updateDocument(IndexWriter.java:1818) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.apache.lucene.index.IndexWriter.addDocument(IndexWriter.java:1471) ~[lucene-core-9.4.2.jar:9.4.2 858d9b437047a577fa9457089afff43eefa461db - jpountz - 2022-11-17 12:56:39]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$Writer.lambda$addIndexMetadataDocuments$5(PersistedClusterStateService.java:1078) ~[main/:?]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$PageWriterOutputStream.flushPage(PersistedClusterStateService.java:1345) ~[main/:?]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$PageWriterOutputStream.close(PersistedClusterStateService.java:1332) ~[main/:?]
  1>    at java.util.zip.DeflaterOutputStream.close(DeflaterOutputStream.java:249) ~[?:?]
  1>    at org.elasticsearch.common.compress.DeflateCompressor$2.close(DeflateCompressor.java:186) ~[main/:?]
  1>    at java.io.FilterOutputStream.close(FilterOutputStream.java:188) ~[?:?]
  1>    at com.fasterxml.jackson.dataformat.smile.SmileGenerator.close(SmileGenerator.java:1894) ~[?:?]
  1>    at org.elasticsearch.xcontent.provider.json.JsonXContentGenerator.close(JsonXContentGenerator.java:567) ~[?:?]
  1>    at org.elasticsearch.xcontent.XContentBuilder.close(XContentBuilder.java:1235) ~[elasticsearch-x-content-8.7.0-SNAPSHOT.jar:8.7.0-SNAPSHOT]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$Writer.writePages(PersistedClusterStateService.java:1106) ~[main/:?]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$Writer.addIndexMetadataDocuments(PersistedClusterStateService.java:1070) ~[main/:?]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$Writer.updateMetadata(PersistedClusterStateService.java:1010) ~[main/:?]
  1>    at org.elasticsearch.gateway.PersistedClusterStateService$Writer.writeIncrementalStateAndCommit(PersistedClusterStateService.java:911) ~[main/:?]
  1>    at org.elasticsearch.gateway.GatewayMetaState$LucenePersistedState.setLastAcceptedState(GatewayMetaState.java:506) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.CoordinationState.handlePublishRequest(CoordinationState.java:392) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.Coordinator.handlePublishRequest(Coordinator.java:464) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.PublicationTransportHandler$PublicationContext$1.get(PublicationTransportHandler.java:361) ~[main/:?]
  1>    at org.elasticsearch.cluster.coordination.PublicationTransportHandler$PublicationContext$1.get(PublicationTransportHandler.java:350) ~[main/:?]
  1>    at org.elasticsearch.action.ActionRunnable$2.accept(ActionRunnable.java:50) ~[main/:?]
  1>    at org.elasticsearch.action.ActionRunnable$2.accept(ActionRunnable.java:47) ~[main/:?]
  1>    at org.elasticsearch.action.ActionRunnable$3.doRun(ActionRunnable.java:72) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:917) ~[main/:?]
  1>    at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[main/:?]
```

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
@@ -70,7 +70,7 @@ public class MaxDocsLimitIT extends ESIntegTestCase {
             .put(super.nodeSettings(nodeOrdinal, otherSettings))
             .put(
                 PersistedClusterStateService.DOCUMENT_PAGE_SIZE.getKey(),
-                ByteSizeValue.ofBytes(randomIntBetween(1000, randomFrom(10000, 100000, 1000000)))
+                PersistedClusterStateService.DOCUMENT_PAGE_SIZE.get(Settings.EMPTY)
             )
             .build();
     }

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
@@ -12,7 +12,9 @@ import org.elasticsearch.action.index.IndexResponse;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.gateway.PersistedClusterStateService;
 import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.index.query.MatchAllQueryBuilder;
 import org.elasticsearch.index.translog.Translog;
@@ -61,6 +63,18 @@ public class MaxDocsLimitIT extends ESIntegTestCase {
         return CollectionUtils.appendToCopy(super.nodePlugins(), TestEnginePlugin.class);
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        // Document page size should not be too small, else we can fail to write the cluster state for small max doc values
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal, otherSettings))
+            .put(
+                PersistedClusterStateService.DOCUMENT_PAGE_SIZE.getKey(),
+                ByteSizeValue.ofBytes(randomIntBetween(1000, randomFrom(10000, 100000, 1000000)))
+            )
+            .build();
+    }
+
     @Before
     public void setMaxDocs() {
         maxDocs.set(randomIntBetween(10, 100)); // Do not set this too low as we can fail to write the cluster state
@@ -72,7 +86,6 @@ public class MaxDocsLimitIT extends ESIntegTestCase {
         restoreIndexWriterMaxDocs();
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/92037")
     public void testMaxDocsLimit() throws Exception {
         internalCluster().ensureAtLeastNumDataNodes(1);
         assertAcked(

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/engine/MaxDocsLimitIT.java
@@ -12,7 +12,6 @@ import org.elasticsearch.action.index.IndexResponse;
 import org.elasticsearch.action.search.SearchResponse;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.unit.ByteSizeValue;
 import org.elasticsearch.common.util.CollectionUtils;
 import org.elasticsearch.gateway.PersistedClusterStateService;
 import org.elasticsearch.index.IndexSettings;

